spectral_cube/spectral_cube.py

```diff
@@ -2420,14 +2422,21 @@ def spatial_smooth_median(self, ksize, **kwargs):
                       self.mask.include(view=(ii, slice(None), slice(None))))
                       for ii in range(self.shape[0]))
 
-        pb = ProgressBar(shape[0])
+        if self.external_update_function is None:
+            pb = ProgressBar(shape[0])
+            update_function = pb.update
+        else:
+            update_function = self.external_update_function
 
         def _gsmooth_image(args):
             """
             Helper function to smooth a spectrum
             """
             (im, includemask),kwargs = args
-            pb.update()
+            if self.abort:
```

Comment:
The current issue with using multi-processing here is that `_gsmooth_image` isn't picklable. I'm concerned that adding a check for a class attribute might make this issue harder to fix in the future. Thoughts @keflavich?